src/libsync/syncengine.cpp

```diff
@@ -992,9 +992,11 @@ void SyncEngine::restoreOldFiles(SyncFileItemVector &syncItems)
             syncItem->_instruction = CSYNC_INSTRUCTION_CONFLICT;
             break;
         case CSYNC_INSTRUCTION_REMOVE:
-            qCWarning(lcEngine) << "restoreOldFiles: RESTORING" << syncItem->_file;
-            syncItem->_instruction = CSYNC_INSTRUCTION_NEW;
-            syncItem->_direction = SyncFileItem::Up;
+            if (syncItem->_type == CSyncEnums::ItemTypeFile) {
```

Comment:
@mgallien don't we also want to restore directories `ItemTypeDirectory`?

Reply:
I updated the logic to allow restoration of folders
thanks